### PR TITLE
add skip redirect param to logout

### PR DESCRIPTION
**Pull request by japfohl**
add parameter allowing user to skip redirect following successful logout

### CODE REVIEW

File: src/api/session.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export interface SessionControl {
   isLoggedIn(): boolean;
   logIn(redirect: true): void;
   logIn(email: string, password: string): Promise<void>;
-  logOut(): void;
+  logOut(skipRedirect: boolean): void;
```
Comment (Crisfole) [inline src/api/session.ts]: `(skipRedirect?: boolean)` makes it optional

File: src/api/session.ts [hunk 2]
```diff
@@ -25,7 +25,7 @@ export function session(settings: ApiSettings): SessionControl {
     isLoggedIn: () => isLoggedIn(settings),
     logIn: (email: string | true, password?: string) => logIn(settings, user.set, email, password),
     renew: (password: string) => renew(settings, user.set, password),
-    logOut: () => logOut(user.set),
+    logOut: (skipRedirect: boolean) => logOut(user.set, skipRedirect),
```
Comment (Crisfole) [inline src/api/session.ts]: ditto

File: src/api/session.ts [hunk 3]
```diff
@@ -88,9 +88,9 @@ function isLoggedIn(settings: ApiSettings) {
   return !!settings.user;
 }
 
-async function logOut(setUser: UserUpdater) {
+async function logOut(setUser: UserUpdater, skipRedirect = false) {
   setUser(null);
-  window.location.href = urls.logOut;
+  skipRedirect && (window.location.href = urls.logOut);
```
Comment (Crisfole) [inline src/api/session.ts]: I'd prefer an `if` here. A bit more explicit. I use `&&` as a boolean operator and as an expression (specifically) never as a statement modifier.

**Review by Crisfole — COMMENTED** [review]
feel free to merge once these are fixed. Release is simply:

`npm version minor`

**Comment by japfohl** [comment]
merged with master

**Pull request merged**